pkg/gateway/contract.go

```diff
@@ -61,3 +64,20 @@ func (c *Contract) SubmitTransaction(name string, args ...string) ([]byte, error
 func (c *Contract) CreateTransaction(name string, args ...TransactionOption) (*Transaction, error) {
 	return newTransaction(name, c, args...)
 }
+
+// RegisterContractEvent registers for chaincode events. Unregister must be called when the registration is no longer needed.
```

Comment:
Just curious why Contract is repeated (struct name + in the func).

Are you planning another type of event?

Reply:
renamed it to RegisterEvent